sdk/azidentity/managed_identity_client.go

```diff
@@ -121,19 +124,29 @@ func (c *managedIdentityClient) createAccessToken(res *azcore.Response) (*azcore
 		}
 		return &azcore.AccessToken{Token: value.Token, ExpiresOn: time.Now().Add(time.Second * time.Duration(expiresIn)).UTC()}, nil
 	}
-	if expiresOn, err := strconv.Atoi(value.ExpiresOn); err == nil {
-		return &azcore.AccessToken{Token: value.Token, ExpiresOn: time.Now().Add(time.Second * time.Duration(expiresOn)).UTC()}, nil
-	}
-	// this is the case when expires_on is a time string
-	// this is the format of the string coming from the service
-	if expiresOn, err := time.Parse("1/2/2006 15:04:05 PM +00:00", value.ExpiresOn); err == nil { // the date string specified is for Windows OS
-		eo := expiresOn.UTC()
-		return &azcore.AccessToken{Token: value.Token, ExpiresOn: eo}, nil
-	} else if expiresOn, err := time.Parse("1/2/2006 15:04:05 +00:00", value.ExpiresOn); err == nil { // the date string specified is for Linux OS
-		eo := expiresOn.UTC()
-		return &azcore.AccessToken{Token: value.Token, ExpiresOn: eo}, nil
-	} else {
-		return nil, err
+	switch v := value.ExpiresOn.(type) {
+	case int:
+		// service fabric is one of the MSI environments that returns an int
+		return &azcore.AccessToken{Token: value.Token, ExpiresOn: time.Now().Add(time.Second * time.Duration(v)).UTC()}, nil
+	case float64:
+		return &azcore.AccessToken{Token: value.Token, ExpiresOn: time.Now().Add(time.Second * time.Duration(v)).UTC()}, nil
+	case string:
+		if expiresOn, err := strconv.Atoi(v); err == nil {
+			return &azcore.AccessToken{Token: value.Token, ExpiresOn: time.Now().Add(time.Second * time.Duration(expiresOn)).UTC()}, nil
```

Comment:
Since we're touching this, can you please also fix this to properly handle Unix time stamp.  See https://github.com/Azure/go-autorest/issues/638 for details.

Reply:
Done...let me know if that's good...I normalized it all to UTC since that's what we were doing before.

Reply:
Actually not done...I saw we're going the same thing for the timestamps....I'll get that fixed right now

Reply:
Ok NO..haha just got mixed up for a second there...those are fine as they are. We're good to go now :P

Reply:
Although when I was debugging in service fabric I saw that the value it returns seems to be something like this: 1.623886173e+09. It falls under the float case....do you know if the the time package handles this properly? Or is some additional parsing necessary?

Reply:
I don't think this is a float, it's just a large number expressed in scientific notation.  Presumably it's the number of seconds from the Unix epoch, thus should be passed as the first arg to [`time.Unix()`](https://pkg.go.dev/time#Unix)

Reply:
I agree that it's not a float, but in the switch case it falls into the float case....but I think you're saying it's fine as I put it then?

Reply:
I'm referring to the int64() cast that time.Unix() expects

Reply:
What is actually sent over the wire?  I suspect this is an artifact of modeling the field as an `interface{}`.

Reply:
I expect it's a pretty large integer that is translated into scientific notation due to it's size. I can spin up the SF cluster again and see if I can get the raw response but it will take some time. I'll also use that to check how the large value is handled with json.Number.

Reply:
So during my testing with json.Number I recalled why I didnt use it, it doesnt accept timestamps only numbers I believe.

Reply:
I looked into this a little, after go 1.14 json.Number is checked a little more strictly during unmarshal...I think the right thing to do is unmarshal into the interface.

Reply:
Ah yes, I do recall something about this being more strict now.

I suppose it's coming back as a `float64` due to this being JSON.

Does this come across the wire as a string?  I ask because regarding your question about losing any significant values, here's what `json.Number` does https://golang.org/src/encoding/json/decode.go?s=7375:7413#L189

Reply:
The cluster keeps stalling right after the GetToken call so havent been able to access the raw value and it's a bit slow to restart the nodes, will try to restart it all again. 
It is definitely not coming back as a string, that's why I had to change the value from string to interface{}, since json.Unmarshal was returning an error about not being able to unmarshal a number into a string.

Reply:
So it returns a value like this: `"expires_on":1624136297`.....that doesnt seem so big to me for it to be parsed into scientific notation....

Reply:
Removed int from the switch case since the docs say that when unmarshalling into an interface json.Numbers will be unmarshalled into a float64.